src/register/pmpaddrx.rs

```diff
@@ -61,7 +61,7 @@ impl PmpAddr {
             // check size is a power of 2
             ((size & (size-1)) != 0) ||
             // checks that the low bits where size is placed, are already zero
-            (addr & encoded_size != 0)
+            ((addr >> 2) & encoded_size != 0)
```

Comment:
Don't you need encoded_size bits and one additional bit to be 0 in addr? The encoding has one 0 followed by 1s.
Example:
yyyy...yy01
yyyy...y011

Reply:
Yea, switched it up to cover that bit too.